plugins/@grouparoo/onesignal/src/lib/export/exportProfile.ts

```diff
@@ -0,0 +1,101 @@
+import { ExportProfilePluginMethod } from "@grouparoo/core";
+import OneSignal from "onesignal-node";
+import { connect } from "../connect";
+
+class InfoError extends Error {
+  errorLevel: string;
+
+  constructor(message) {
+    super(message);
+    this.errorLevel = "info";
+  }
+}
+
+export const exportProfile: ExportProfilePluginMethod = async ({
+  appOptions,
+  export: {
+    newProfileProperties,
+    oldProfileProperties,
+    newGroups,
+    oldGroups,
+    toDelete,
+  },
+}) => {
+  if (Object.keys(newProfileProperties).length === 0) {
+    return { success: true };
+  }
+
+  const client = await connect(appOptions);
+
+  const extUserId = newProfileProperties["external_user_id"];
+  if (!extUserId) {
+    throw new Error(
+      `newProfileProperties[external_user_id] is a required mapping`
+    );
+  }
+
+  const payload: any = { tags: {} };
+
+  // set profile properties, including old ones.
+  let newKeys = Object.keys(newProfileProperties);
+  const oldKeys = Object.keys(oldProfileProperties);
+  const allKeys = new Set([...newKeys, ...oldKeys]);
+
+  if (toDelete) {
+    // When deleting, we just clear the old tags. No real deletion occurs.
+    newKeys = [];
+    newGroups = [];
+  }
+
+  for (const key of allKeys) {
```

Comment:
do we think keys need to be normalized too?

Reply:
Yeah it seems like you can set any string as the tag key, but their docs recommend to... 

![image](https://user-images.githubusercontent.com/4368928/110734360-09249200-81fe-11eb-941c-258a4c79d1ff.png)

While it does work, I think normalizing to be better safe than sorry could be a good option

Reply:
cool. let's do that then with the same function as the groups.